### ISSUE TITLE
ENTITY_IS_DELETED error on Informatica integration

### ISSUE THREAD

**Issue by ceiroa** [SalesforceFoundation/HEDAP#333]
From https://powerofus.force.com/_ui/core/userprofile/UserProfilePage?u=0058000000BMwyH&tab=sfdc.ProfilePlatformFeed&fId=0D580000032FgmD.

After version 1.19 was pushed to production on 27/10/16, user reported getting the following error from an existing Informatica integration that was previously working correctly:

Error Type: System.DmlException
Error Date: 2016-10-27 11:50:48
Message: "Delete failed. First exception on row 0 with id a2R4E0000001L5UUAU; first error: ENTITY_IS_DELETED, entity is deleted: []"
 
Stack Trace:
Stacktrace available at the time of saving the record - Class.hed.TDTM_Config.setTdtmConfig: line 82, column 1
Class.hed.TDTM_Config.getClassesToCallForObject: line 124, column 1
Class.hed.TDTM_TriggerHandler.run: line 75, column 1
Class.hed.TDTM_Global_API.run: line 60, column 1
Trigger.hed.TDTM_Account: line 33, column 1

**Comment by ceiroa** [comment]
**lurch: add

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x800000049kD3AAI" target="_blank">W-011737</a>

**Comment by ceiroa** [comment]
I was able to reproduce the issue with a script like this:

```
require 'restforce'

def api_client
  @api_client ||= Restforce.new api_version: '32.0',
                                refresh_token: 'xxxx',
                                client_id: 'xxxx',
                                client_secret: 'xxxx'
end

def create_account_via_api
  @account_id = @api_client.create('Account', Name: 'foo')
end

api_client

t1 = Thread.new do
	create_account_via_api
end
create_account_via_api
t1.join
```

**Comment by jlantz** [comment]
Included in beta release 1.20 (Beta 1)

**Comment by mrbelvedere** [comment]
Included in production release rel/1.20

**Comment by ceiroa** [comment]
@njjc ~ See error message from client with full stack trace.
